tests/filecheck/dialects/csl/csl-wrapper-ops.mlir

```diff
@@ -0,0 +1,99 @@
+// RUN: XDSL_ROUNDTRIP
+// RUN: XDSL_GENERIC_ROUNDTRIP
+
+builtin.module {
+    csl_wrapper.module() <{"width"=10 : i16, "height"=10: i16, "z_dim"=10: i16, "pattern"=5: i16}> ({
```

Comment:
> Is this referring to the custom format or prop_defs on the op?

Yes, I find this custom syntax highly disorienting, as it appears to render the ops properties (enclosed by `<{`, but in actuality displays a weird mix of properties and the `params` property. The generic syntax below feels much more readable and clear than the custom syntax introduced here.

> I can't seem to think of a use-case where we'd ever want to pass something not int as a param to both modules.

How would we pass a `memref`? or a list of weights? or an enum to select a mode?

I think this is an overgeneralisation that will hinder us in the near future.

Reply:
> > Is this referring to the custom format or prop_defs on the op?
> 
> Yes, I find this custom syntax highly disorienting, as it appears to render the ops properties (enclosed by `<{`, but in actuality displays a weird mix of properties and the `params` property. The generic syntax below feels much more readable and clear than the custom syntax introduced here.

Fair enough, will change it.

> > I can't seem to think of a use-case where we'd ever want to pass something not int as a param to both modules.
> 
> How would we pass a `memref`? or a list of weights? or an enum to select a mode?

I think these can't actually be passed as csl.params, i.e. from command line to the layout module, and I *think* we also can't pass buffers from layout.csl to program.csl.

It's actually a really good point you're making, let me expand - the translation plan is to generate one csl wrapper per module-level func.func. Since funcs are isolated from above, the func args and results effectively represent the memrefs/buffers/etc that can be transferred between host and device. They wouldn't be lowered to csl `param`s and need to be exported. Right now, the func is currently simply dumped in the `csl_wrapper.program_module` body. I'd like the module wrapper to keep track of various types of things we will need to refer to in lowering:

* params (proposed in this PR)
* exported symbols allowing host-device transfers corresponding to func args/results (not done in this PR, though rough idea how to)
* program globals not exported (not really sure how to manage them in a way they can be easily accessed or whether there even is a need to)
* exported program entry point corresponding to func name (also not done in this PR)

Reply:
I would suggest having params as something like `params = [#hlcsl.param<"z_dim", default=3: i32>, #hlcsl.param<"pattern" : i32>, ...]`

The attribute would be:

```py
@irdl_attr_definition
class HlCslParam(ParametrizedAttribute):
    name = "hlcsl.param"

    key: ParamAttr[StringAttr]
    value: ParamAttr[AnyIntegerAttr | NoneAttr]
    type: ParamAttr[IntegerType]
```

Reply:
Done